### PR TITLE
implement search-by-id pkg test

### PR DESCRIPTION
**Pull request by kmrmt**
Signed-off-by: Kosuke Morimoto <ksk@vdaas.org>

### Description:
SSIA

### Environment:

- Go Version: 1.18.1
- Docker Version: 20.10.8
- Kubernetes Version: 1.22.0
- NGT Version: 1.14.3

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [X] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [X] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [X] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pkg/agent/core/ngt/handler/grpc/handler_test.go [hunk 1]
```diff
@@ -1312,6 +1312,46 @@ func Test_server_SearchByID(t *testing.T) {
...
+	/*
+		SearchByID test cases:
+		- Equivalence Class Testing
+			- case 1.1: success search vector from 1000 vectors (type: uint8)
+			- case 1.2: success search vector from 1000 vectors (type: float32)
+			- case 2.1: fail search with non-existent ID form 1000 vectors (type: uint8)
+			- case 2.2: fail search with non-existent ID form 1000 vectors (type: float32)
+		- Boundary Value Testing
+			- case 1.1: success search with "" from 1000 vectors (type: uint8)
+			- case 1.2: success search with "" from 1000 vectors (type: float32)
```
Comment (kevindiu) [inline pkg/agent/core/ngt/handler/grpc/handler_test.go:1325]: I am thinking do we need both uint8 and float32 cases as the test case is focus on testing the ID instead of the vector type

File: pkg/agent/core/ngt/handler/grpc/handler_test.go [hunk 2]
```diff
@@ -1312,6 +1312,46 @@ func Test_server_SearchByID(t *testing.T) {
...
 	}
+
+	/*
+		SearchByID test cases:
+		- Equivalence Class Testing
+			- case 1.1: success search vector from 1000 vectors (type: uint8)
+			- case 1.2: success search vector from 1000 vectors (type: float32)
+			- case 2.1: fail search with non-existent ID form 1000 vectors (type: uint8)
+			- case 2.2: fail search with non-existent ID form 1000 vectors (type: float32)
+		- Boundary Value Testing
```
Comment (vankichi) [inline pkg/agent/core/ngt/handler/grpc/handler_test.go:1323]: How about removing `from 1000 vectors` from each test case and showing it here instead?
e.g.
```suggestion
		- Boundary Value Testing ( 1000 vectors inserted before a search )
```

**Comment by kmrmt** [comment]
I implement Boundary Value Testing case 3.x  and fix the name for real situations.

And I have a question about Boundary Value Testing case 1.1.
It cannot insert an empty string ID into Vald, so it cannot search an empty string ID as test.

**Review by vankichi — APPROVED** [review]

**Review by kevindiu — APPROVED** [review]
